### ISSUE TITLE
remove_all_* does not trigger hooks

### ISSUE THREAD

**Issue by Gibheer** [jeremyevans/sequel#931]
Hi,

when using the after_remove hook, the remove_all_* functions do not trigger these hooks. See the example code.

```ruby
require 'sequel'

DB = Sequel.sqlite # memory database

DB.create_table :carts do
  primary_key :id
  Text :name
end

DB.create_table :items do
  primary_key :id
  foreign_key :cart_id, :carts
  Text :name
end

def log_remove(*args)
  puts "deleting: #{args.inspect}"
end

class Cart < Sequel::Model;
  one_to_many :items, :after_remove => :log_remove
end
class Item < Sequel::Model;
  many_to_one :cart, :after_remove => :log_remove
end

# Populate the table
cart = DB[:carts].insert(:name => 'foo')
DB[:items].insert(:name => 1, :cart_id => cart)
DB[:items].insert(:name => 2, :cart_id => cart)
DB[:items].insert(:name => 3, :cart_id => cart)

Cart.first.remove_item(1) # produces a log entry
Cart.first.remove_all_items # nothing

**Comment by jeremyevans** [comment]
This is expected behavior, but I'm not sure if it is documented.  I'll add the appropriate documentation.

If you want the hooks to be called and remove all associated objects:  `c = Cart.first; c.items.each{|i| c.remove_item(i)}`

**Comment by jeremyevans** [comment]
Fixed by  2ab5ed8
